include/ck_tile/ref/naive_grouped_conv_fwd.hpp

```diff
@@ -0,0 +1,319 @@
+// SPDX-License-Identifier: MIT
+// Copyright (c) 2025, Advanced Micro Devices, Inc. All rights reserved.
+
+#pragma once
+
+#include "ck_tile/core.hpp"
+#include "ck_tile/host/device_memory.hpp"
+#include "ck_tile/host/kernel_launch.hpp"
+#include <hip/hip_runtime.h>
+
+namespace ck_tile {
+
+// Naive GPU reference kernel struct for forward grouped convolution
+// Layout: Input=NDHWGC, Weight=GKZYXC, Output=NDHWGK (for 3D case)
+//         Input=NHWGC,  Weight=GKYXC,  Output=NHWGK  (for 2D case)
+//         Input=NWGC,   Weight=GKXC,   Output=NWGK   (for 1D case)
+//
+// One thread per output element, uses grid-stride loop pattern
+
+template <ck_tile::index_t NDimSpatial,
+          typename InDataType,
+          typename WeiDataType,
+          typename OutDataType>
+struct naive_grouped_conv_fwd_kernel
+{
+    static constexpr ck_tile::index_t kBlockSize = 256;
+
+    __device__ void operator()(const InDataType* __restrict__ p_in,
+                               const WeiDataType* __restrict__ p_wei,
+                               OutDataType* __restrict__ p_out,
+                               // Tensor dimensions
+                               ck_tile::index_t G, // number of groups
+                               ck_tile::index_t N, // batch size
+                               ck_tile::index_t K, // output channels per group
+                               ck_tile::index_t C, // input channels per group
+                               // Input spatial dimensions
+                               const ck_tile::long_index_t* in_spatial_lengths,
+                               // Weight spatial dimensions
+                               const ck_tile::long_index_t* wei_spatial_lengths,
+                               // Output spatial dimensions
+                               const ck_tile::long_index_t* out_spatial_lengths,
+                               // Convolution parameters
+                               const ck_tile::long_index_t* conv_strides,
+                               const ck_tile::long_index_t* conv_dilations,
+                               const ck_tile::long_index_t* in_left_pads) const
+    {
+        const ck_tile::long_index_t tid         = blockIdx.x * blockDim.x + threadIdx.x;
```

Comment:
We have some functions which you can use for this: https://github.com/ROCm/composable_kernel/blame/develop/include/ck_tile/core/arch/arch.hpp#L101-L119